Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/local_profile_model.ts

```diff
@@ -38,6 +38,17 @@ export class LocalProfileModel extends ProfileModel {
     return this._roles.clone();
   }
 
+  public get groups(): Beam.DirectoryEntry[] {
+    if(!this.isLoaded) {
+      throw Error('Model not loaded.');
+    }
+    return this._groups;
+  }
+
+  public set groups(groups:Beam.DirectoryEntry[]) {
+    this._groups = groups;

```

Comment:
Make a copy, once again this prevents problems where a modification to ```groups``` ends up affecting the model.